### ISSUE TITLE
Running airflow upgrade_check failed on DatabaseVersionCheckRule with postgres images

### ISSUE THREAD

**Issue by alonrolnik** [apache/airflow#16188]
**Apache Airflow version**: 1.10.15

**Environment**:

**What happened**:

Running airflow upgrade_check failed on DatabaseVersionCheckRule with postgres image, for example, postgres:13.3

```
(app-root) sh-4.4$ airflow upgrade_check
steps_version_dict: {'sample-step-service': 'latest', 'aws-training-step': 'latest', 'aws-preprocess-step': 'latest', 'aws-forecast-step': 'latest', 'csv-cleanup-step': 'latest', 'customer-data-export-step': 'latest', 'customer-data-import-step': 'latest', 'forecasting-statistics-step': 'latest', 'aws-predictor-delete-step': 'latest', 'aws-multi-predictor-delete-step': 'latest', 'aws-customer-delete-trigger-step': 'latest', 's3-delete-step': 'latest', 'customer-db-delete-step': 'latest', 'flow-status-step': 'latest', 'airflow-sidecar': 'local_dev'}
Using: docker-unstable.anaplan-np.net/planning-ai/planning-ai-airflow-sidecar:local_dev for airflow sidecar

=============================================================================== STATUS ==============================================================================

Check for latest versions of apache-airflow and checker....................................................................................................SUCCESS
Remove airflow.AirflowMacroPlugin class....................................................................................................................SUCCESS
Ensure users are not using custom metaclasses in custom operators..........................................................................................SUCCESS
Chain between DAG and operator not allowed.................................................................................................................SUCCESS
Connection.conn_type is not nullable.......................................................................................................................SUCCESS
Custom Executors now require full path.....................................................................................................................SUCCESS
Traceback (most recent call last):
  File "/opt/app-root/bin/airflow", line 37, in <module>
    args.func(args)
  File "/opt/app-root/lib/python3.6/site-packages/airflow/upgrade/checker.py", line 118, in run
    all_problems = check_upgrade(formatter, rules)
  File "/opt/app-root/lib/python3.6/site-packages/airflow/upgrade/checker.py", line 38, in check_upgrade
    rule_status = RuleStatus.from_rule(rule)
  File "/opt/app-root/lib/python3.6/site-packages/airflow/upgrade/problem.py", line 44, in from_rule
    result = rule.check()
  File "/opt/app-root/lib/python3.6/site-packages/airflow/utils/db.py", line 74, in wrapper
    return func(*args, **kwargs)
  File "/opt/app-root/lib/python3.6/site-packages/airflow/upgrade/rules/postgres_mysql_sqlite_version_upgrade_check.py", line 52, in check
    installed_postgres_version = Version(session.execute('SHOW server_version;').scalar())
  File "/opt/app-root/lib/python3.6/site-packages/packaging/version.py", line 298, in __init__
    raise InvalidVersion("Invalid version: '{0}'".format(version))
packaging.version.InvalidVersion: Invalid version: '13.3 (Debian 13.3-1.pgdg100+1)'
```
**What you expected to happen**:
I expect the check to pass.

**How to reproduce it**:
Spin up postgres container from postgres repository and run `airflow upgrade_check`

**Comment by potiuk** [comment]
Yeah. Bad check. The reported version seems to be OK so you can proceed.

I am not sure though if we will release a new upgrade check version. 

As a quick workaround, you can simply disable the `DatabaseVersionCheckRule` as described here:

https://airflow.apache.org/docs/apache-airflow/stable/upgrade-check.html#turning-off-checks

**Comment by eladkal** [comment]
It's fixed in https://github.com/apache/airflow/pull/15122

@potiuk we have several PRs already merged for Airflow Upgrade Check - 1.4.0 including a fix for this one :)

**Comment by potiuk** [comment]
Ah cool !
